bot/cogs/moderation/modlog.py

```diff
@@ -116,6 +118,21 @@ def ignore(self, event: Event, *items: int) -> None:
 
         return await self.bot.get_context(log_message)  # Optionally return for use with antispam
 
+    async def reupload_attachments(
+            self,
+            message: discord.Message,
+            channel_id: int = GuildConstant.attachment_repost
+    ) -> t.List[str]:
+        """Re-upload message's attachments to the the channel_id and return the list of re-posted attachments URLs."""
+        channel = self.bot.get_channel(channel_id)
+        out = []
+        for attachment in message.attachments:
+            with BytesIO() as buffer:
+                await attachment.save(buffer, use_cached=True)
```

Comment:
Shouldn't there be some error handling for failed saves? According to the docs, the `proxy_url` may still fail.

Reply:
Well I'm going to move this function so there shouldn't be a case where the `proxy_url` fail, but if we get a traceback, we don't really care, because we just abort the re-upload.

Reply:
I would think it's better to try to upload as many attachments as possible rather than abort when any fail.

Reply:
Anyway, the re-upload will take place before the message deletion, so there shouldn’t be any reason for the re-upload to fail, and if there is, we should better log it and continue re-uploading isn’t?

Reply:
Yeah, just a simple log is fine.